assets/js/workflow-editor/index.ts

```diff
@@ -94,19 +124,58 @@ export default {
     // between the current state and the server state and send those diffs
     // to the server.
   },
+  setupObserver() {
+    this.observer = new MutationObserver(mutations => {
+      mutations.forEach(mutation => {
+        const { attributeName, oldValue } = mutation as AttributeMutationRecord;
+
+        if (attributeName == 'data-base-url') {
+          const newValue = this.el.getAttribute(attributeName);
+
+          if (oldValue !== newValue) {
+            this.baseUrl = newValue;
+          }
+        }
+      });
+    });
+
+    this.observer.observe(this.el, {
+      attributeFilter: ['data-base-url'],
+      attributeOldValue: true,
+    });
+  },
+  getItem(id?: string) {
+    if (id) {
+      const { jobs, triggers, edges } = this.workflowStore.getState();
+      const everything = [...jobs, ...triggers, ...edges];
+      for (const i of everything) {
+        if (id === i.id) {
+          return i;
+        }
+      }
+    }
+  },
   onSelectionChange(id?: string) {
-    const currentUrl = new URL(window.location.href);
+    const currentUrl = new URL(this.baseUrl!);
```

Comment:
@josephjclark note this change, I'm keeping a copy of the url from `handle_params` in the liveview, and passing it to the component... it seems that between liveviews dom diffing and the browsers history patch updates, the component _can_ mount and fire a patch _before_ (and therefore based on an old url) `window.location` has been updated. 

Resulting in a url like `.../w?s=...` so you never actually see the new workflow template.